scipy/stats/_stats_py.py

```diff
@@ -83,20 +83,15 @@
            'brunnermunzel', 'alexandergovern']
 
 
-# This should probably be rewritten to avoid nested TypeErrors in favor of
-# branching based on dtype.
 def _contains_nan(a, nan_policy='propagate', use_summation=True):
     if not isinstance(a, np.ndarray):
         use_summation = False  # some array_likes ignore nans (e.g. pandas)
     policies = ['propagate', 'raise', 'omit']
     if nan_policy not in policies:
         raise ValueError("nan_policy must be one of {%s}" %
                          ', '.join("'%s'" % s for s in policies))
-    # only inexact (floating/complexfloating) and object arrays support np.nan
-    if not (np.issubdtype(a.dtype, np.inexact)
-            or np.issubdtype(a.dtype, object)):
-        return False, nan_policy
-    try:
+
+    if np.issubdtype(a.dtype, np.number):
```

Comment:
```suggestion
    if np.issubdtype(a.dtype, np.inexact):
```
Exact numbers (e.g. `int`) can't store NaN.